package/tvheadend-app-osmc/files/DEBIAN/postinst

```diff
@@ -1,33 +1,86 @@
-#!/bin/bash
+#! /bin/bash
 
 if [ "$1" = "configure" ]; then
 
 	systemctl daemon-reload
 
-	if [ ! -f /home/osmc/.hts/tvheadend/accesscontrol/1 ] && [ -e "/var/run/${DPKG_MAINTSCRIPT_PACKAGE}_install" ]; then
+	# check for existing user called osmc
+	OSMCUSERFILE=$(grep -l "\"username\": \"osmc\"" /home/osmc/.hts/tvheadend/accesscontrol/* )
+	echo userfile is: $OSMCUSERFILE
 
-		echo "Creating TVHeadend default access control file"
-		mkdir -p /home/osmc/.hts/tvheadend/accesscontrol
+	if [ -z "$OSMCUSERFILE" ]; then
```

Comment:
Just a suggestion.

We only need some of those migration steps from 4.0.9 I suspect.
You don't want to do them when going from 4.2.2 to 4.2.3 etc, correct?